docs/developer/update-v1alpha1-provider-to-v1alpha2-infra-provider.md

```diff
@@ -0,0 +1,188 @@
+
+### The new API group
+
+Cluster API v1alpha2 custom resource definitions (CRDs) belong to the API group `cluster.x-k8s.io`.
+
+### The infrastructure API group
+
+An infrastructure provider's CRDs should belong to the API group `infrastructure.cluster.x-k8s.io`.
+
+### Kubebuilder
+
+While [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) v2 is available, the recommended approach for updating a CAPI provider to v1alpha2 is to stick with kubebuilder v1 during the update process and then reevaluate kubebuilder v2 after a successful migration to CAPI v1alpha2.
+
+### Sample code and other examples
+
+This document uses the CAPI provider for AWS ([CAPA](https://github.com/kubernetes-sigs/cluster-api-provider-aws)) for sample code and other examples.
+
+## Create a branch for new v1alpha1 work
+
+This document assumes the work required to update a provider to v1alpha2 will occur on the project's `master` branch. Therefore, the recommendation is to create a branch `release-MAJOR.MINOR` in the repository from the latest v1alpha1-based release. For example, if the latest release of a provider based on CAPI v1alpha1 was `v0.4.1` then the branch `release-0.4` should be created. Now the project's `master` branch is free to be a target for the work required to update the provider to v1alpha2, and fixes or backported features for the v1alpha1 version of the provider may target the `release-0.4` branch.
+
+## Update the API group in the `PROJECT` file
+
+Please update the `PROJECT` file at the root of the provider's repository to reflect the API group `cluster.x-k8s.io`:
+
+```properties
+version: "1"
+domain: cluster.x-k8s.io
+repo: sigs.k8s.io/cluster-api-provider-aws
+```
+
+## Create the provider's v1alpha2 resources
+
+The new v1alpha2 types are located in `pkg/apis/infrastructure/v1alpha2`.
+
+### The cluster and machine resources
+
+Providers no longer store configuration and status data for clusters and machines in the CAPI `Cluster` and `Machine` resources. Instead, this information is stored in two, new, provider-specific CRDs:
+
+* `pkg/apis/infrastructure/v1alpha2.`_Provider_`Cluster`
+* `pkg/apis/infrastructure/v1alpha2.`_Provider_`Machine`
+
+For example, the AWS provider defines:
+
+* [`sigs.k8s.io/cluster-api-provider-aws/pkg/apis/infrastructure/v1alpha2.AWSCluster`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6de25b31def9b4203a3c0a92b868a1819ea6e3e7/pkg/apis/infrastructure/v1alpha2/awscluster_types.go#L138-L146)
+* [`sigs.k8s.io/cluster-api-provider-aws/pkg/apis/infrastructure/v1alpha2.AWSMachine`](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6de25b31def9b4203a3c0a92b868a1819ea6e3e7/pkg/apis/infrastructure/v1alpha2/awsmachine_types.go#L144-L152)
+
+### The spec and status types
+
+The `Spec` and `Status` types used to store configuration and status information are effectively the same in v1alpha2 as they were in v1alpha1:
```

Comment:
Would it be good to call out that the kubeadm related configuration and CA certificate info has been extracted?